app/com/linkedin/drelephant/configurations/tunin/TuninAlgorithmConfiguration.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2016 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.linkedin.drelephant.configurations.tunin;
+
+import com.linkedin.drelephant.util.Utils;
+import org.apache.log4j.Logger;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public class TuninAlgorithmConfiguration {
+
+    private static final Logger logger = Logger.getLogger(TuninAlgorithmConfiguration.class);
+    private List<TuninAlgorithmConfigurationData> _tuninAlgorithmConfDataList;
+
+    public TuninAlgorithmConfiguration(Element configuration) {
+        parseTuninAlgorithmConfiguration(configuration);
+    }
+
+    /**
+     * Returns the list of Tuning algorithms with their Configuration Information
+     *
+     * @return A list of Configuration Data for tuning
+     */
+    public List<TuninAlgorithmConfigurationData> getTuninAlgorithmsConfData() {
```

Comment:
Who uses this function ? How do you make sure they dont write to the list ?

Reply:
This is not in use for now. Added for complete configuration driven logic.

Reply:
This class has been removed